enhancements/running-on-the-host.md

```diff
@@ -0,0 +1,167 @@
+
+## Non Goals
+
+- Having an alternative way to provide the static configuration that involves distritbuting the configuration to the routers
+
+## Proposal
+
+### User stories
+
+- As a cluster admin, I want to deploy the router to the nodes of the cluster with enough configuration to allow it to provide the basic connectivity.
+
+- As a cluster admin, I want to use Kubernetes CRDs to provision additional networks to reach / expose different classes of services or implement multitenancy.
+
+### Design Details
+
+The general idea is to run the same set of containers we have in pods today, but on the host. This includes:
+
+- running the router pod, in charge of providing the network control plane with FRR and the data plane with veths and the underlay interface
+- running the controller pod, in charge of reading the static configuration and receiving the additional configuration when the kubernetes api is available
+
+#### Handling the lifecycle of the components
+
+The router and the controller need to run as systemd units. This allows us to run them before the kubelet and the CNI start.
+Once the machinery is implemented, alternative ways to run them can be added for those environments where systemd is not enabled.
+
+#### Running on the host
+
+Podman provides a way to maintain the current pod structure while running the pods as systemd units. This allow us to keep reasoning of pods and
+to have a good parallel between the "Kubernetes mode" and the "Host mode".
+
+Because of this, the initial implementation will be based on Podman. As per systemd above, when we have a way of running the pods on the host, it should be
+relatively easy to adapt to other container runtimes.
+
+#### Interactions
+
+There are some interactions between the controller process and the router (and the surrounding system) that need to be adapted to the new scenario:
+
+##### Retrieving the router's network namespace
+
+We are using the container runtime api to retrieve the network namespace of the pod.
+When running on the host, the target namespace can be reitrieved by telling podman to fill a pid file, and by using the pid to retrieve the network namespace of the process.
+
+##### Sending a signal to the reloader container to reload the FRR configuration
+
+Currently the controller retrieves the target pod's IP using the kubernetes API. When running on the host, a unix socket can be shared between the containers.
+
+##### Restarting the router pod
+
+The controller pod currently deletes the router pod to restart the underlay creation machinery. When running on the host, the controller pod can interact either with Podman or with Systemd to restart the pod.
+
+#### Consuming the static configuration
+
+The static configuration must be provided to the OpenPERouter as files in a known location. The configuration is composed by:
+
+- A static configuration with the same structure of the CRDs (ie underlays, vnis) to be used for bringing up the basic connectivity
```

Comment:
It is just the structure or it will be exactly the same yamls ? can you show an example in case it's different ? I think it can be directly the same yaml even with the kubevirt bits, so users need to understand just one type and it can be consumed directly as kube api provided resource.

Reply:
I kind of like the GRPC approach, although I see it split the controller.